util/provenance/buildconfig.go

```diff
@@ -0,0 +1,170 @@
+package provenance
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/moby/buildkit/solver"
+	"github.com/moby/buildkit/solver/pb"
+	digest "github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+)
+
+type BuildConfig struct {
+	Definition []BuildStep `json:"llbDefinition,omitempty"`
+}
+
+type BuildStep struct {
+	ID     string      `json:"ID,omitempty"`
```

Comment:
Should `ID` be in caps? [SLSA provenance](https://slsa.dev/provenance/v0.2) looks like it prefers `id`.